picby_app/src/views/auth/forgotPassScreen.tsx

```diff
@@ -40,110 +43,152 @@ type Props = {
   navigation: NavigationStackProp;
 };
 
-const ForgotPasswordScreen: React.FC<Props> = ({navigation: {navigate}}) => {
+interface InputValueType {
+  email: string;
+}
+
+interface ActionTypes {
+  resetForm: () => void;
+}
+
+const ForgotPasswordScreen: React.FC<Props> = ({navigation}) => {
+  const {navigate} = navigation;
   const {handlePopUpAnimation, fadeAnim} = useHandlePopupAnimation();
-  const {dismissKeyboard} = useContext(AuthContext);
-  const [emailNotFoundError, setEmailNotFoundError] = useState(false);
-  const [serverResponseStatus] = useState(true);
+  const {
+    dismissKeyboard,
+    forgotPassContextData: {
+      isEmailNotFound,
+      isItForgotPassServerError,
+      areForgotPassButtonsDisabled,
+      setAreForgotPassButtonsDisabled,
+      setIsEmailNotFound,
+      isEmailSendSuccess,
+      handleForgotPasswordRequestAndErrors,
+      setForgotScreenStateToDefault,
+    },
+  } = useContext(AuthContext);
   const {
     messageBadMail,
     messageEmailNotFound,
-    popUpText,
+    messageSendSuccess,
     contentText,
     contentHeader,
+    messageServerError,
   } = forgotPasswordMessages;
   const {placeholderTextBlueColor} = inputData;
   const {textColorWhite, textColorBlue, sendText, goBackText} = buttonsData;
 
-  const handleSendEmailRequest = () => {
-    let promise = new Promise((res, rej) =>
-      setTimeout(() => {
-        serverResponseStatus ? res(true) : rej(true);
-      }, 3000),
-    );
-    return promise
-      .then(() => {
-        handlePopUpAnimation();
-      })
-      .catch(() => {
-        setEmailNotFoundError(true);
-      });
-  };
+  const [messagePopUpText, setMessagePopUpText] = useState('');
+
+  React.useEffect(() => {
```

Comment:
używamy tego bez dependency array? To znaczy że kiedy ten efekt się będzie trigerował?